### ISSUE TITLE
Consistent styling between matplotlib and bokeh backends

### ISSUE THREAD

**Issue by jbednar** [ioam/holoviews#1339]
At least as an option, it would be nice if there were consistent styling defaults between the two main backends.  Off the top of my head:

- [ ] Default plot sizes differ (such that 4 subplots usually fit for mpl, but 3 for bokeh)
- [ ] Default background colors differ (or they used to?)
- [ ] mpl backend shows subfigure labels by default, but bokeh does not support them (so they would first have to be added, then enabled by default, before the two match.
- [ ] Are there different colormaps/color cycles in some cases?  I think there used to be.

Please edit this list to include any other obvious differences.

If we agree this should be done, I would assume it should be in 2.0.

**Comment by jlstevens** [comment]
I could imagine the notebook extension being the place to select such a 'theme'. I don't see why you would want to change from 'hv default' because anything else would make working between backends more difficult than necessary. Instead I imagine just two options, one for consistency between matplotlib and bokeh (that might keep being improved over time) and one for backwards compatibility.

Thinking about this I think here are two things we should aim for:

1. Getting HoloViews plot options to behave the same way between backends. We already do this but there are certainly areas where we could improve.
2. Picking good style option defaults for each backend so they match as closely as we can.

Anything else is probably outside our core design where we pass style keywords from the user straight to the applicable backend. I think we should try to work on these types of improvement for version 2.0, even if we only get a chance to target the easy fixes.

**Comment by jbednar** [comment]
I agree the notebook extension is a good place to choose such a theme.  I guess you're proposing a fourth theme, "hv compatibility" or "hv static", i.e. one that does not change over time.  The reason I suggested having "mpl 2.0" and "bokeh 1.0" is to make it simpler to mix and match between hv and native mpl, or hv and native bokeh. If people primarily use hv as an addon to their existing mpl workflow, then they probably don't care about whether the styling matches when they switch to bokeh, since they probably won't be switching to bokeh.  Instead, they care about whether it matches their other non-hv mpl plots.  Similarly and vice versa for bokeh users augmenting with hv plots (e.g. to use datashader more easily alongside their native Bokeh plots); there's no reason they would care about matching mpl, but lots of reasons to want to match pure Bokeh.

**Comment by sverley** [comment]
This is something I don't understand about holoviews. Reading the goals of this module, it aims to make your graph renderer agnostic. But actually, you need to use different options name to set style and so on if you change your renderer. Is it so hard to make them compatible? I would definitively use only hv if it offer that service.

My last use case was a notebook I used to analyse and understand my data. For that, I use bokeh extension since plots are dynamics and curves can be muted (I haven't found a solution with mpl). So far, I need to save all my plots as image files but I wasn't able to save them with bokeh due to browser driver error on which I cannot do anything since the host is administrated by IT. So I switch to mpl to save them but then, I don't get the plots I set up.

**Comment by jlstevens** [comment]
@sverley The way holoviews works is that we try to make a whole lot of plot building options as consistent as we can across backends (plot options) while styling options get passed through to the backend. There the differences between the backends are quite apparent and yes, keeping a mapping (and introducing some sort of middle-of-the-road API) would be a big task and not one I think we could ever fully succeed at.

However, [hvplot](https://hvplot.holoviz.org/) is the API where we have been working on such a mapping to try and generate style options consistent between backends (very much still in progress!). I think we should consider how to make this mapping easily available to HoloViews users once it is in a more complete state.

**Comment by sverley** [comment]
In fact, I was using `hvplot` to generate my plots but I've needed to use `hv` to manipulate them. Surprisingly, I was setting plot style using `hv` and not `hvplot`... Thanks, it is much better. Altough, I've tried to use `hvplot.render` in order to save my plots using matplotlib but since the plot has been generated before with bokeh options, I fallback to the same behaviour as using hv directly. Unfortunately, it seems like `compatibility='bokeh'` only works for `hvplot.extension`. Any idea to make output from one backend to another from plot object?

```python
import pandas as pd
import numpy as np
import hvplot
import hvplot.pandas
hvplot.extension('bokeh')

def lissajous(t, a, b, delta):
    return (np.sin(a * t + delta), np.sin(b * t), t)

lin = np.linspace(0, np.pi*2, 200)
data = pd.DataFrame({v:d for v,d in zip(('x','y','t'),lissajous(lin, 3, 5, np.pi/2))})

p = data.hvplot(x="x",y="y", line_dash="dashed", height=400, width=400, title="Title")
p
```
![bokeh_plot](https://user-images.githubusercontent.com/8038190/214872128-c06c1f30-7192-415a-99cb-31fed709b1fd.png)
```python
hvplot.output(p, backend="matplotlib")
```
![image](https://user-images.githubusercontent.com/8038190/214874388-352a1367-b95f-41e5-836e-9b3257d93e1e.png)
If now I set extension to `matplotlib`, I have:
```python
hvplot.extension('matplotlib')
p = data.hvplot(x="x",y="y", line_dash="dashed", height=400, width=400, title="Title")
p
```
![image](https://user-images.githubusercontent.com/8038190/214874985-8a01bb42-dae0-413e-a3e2-10afda695dbe.png)
```python
hvplot.output(p, backend="bokeh")
```
![bokeh_plot(2)](https://user-images.githubusercontent.com/8038190/214875189-2e53aaf5-7fb9-4be0-8e04-84882fd55370.png)
And now if I add `compatibility='bokeh'`:
```python
hvplot.extension('matplotlib', compatibility='bokeh')
p = data.hvplot(x="x",y="y", line_dash="dashed", height=400, width=400, title="Title")
p
```
![image](https://user-images.githubusercontent.com/8038190/214876231-5439b865-2588-4312-8cdf-b1207fb76170.png)
```python
hvplot.output(p, backend="bokeh")
```
![bokeh_plot(3)](https://user-images.githubusercontent.com/8038190/214876200-b4870cba-faa8-4541-b53e-59456d1560cb.png)

The behaviour of this is strange since not all options are passing from one plot to another. For example, `title` pass from `matplotlib` to `bokeh` but the other way.
